GoogleInstagram/app/src/main/java/com/example/cynchen/googleinstagram/SearchFragment.java

```diff
@@ -0,0 +1,175 @@
+        //Three: All other j index values -> load the (j+1)th image
+        nextbutton = (Button) rootView.findViewById(R.id.next_image);
+        nextbutton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (array_images.size() == 0) {
+                    create_alert_search();
+                }
+                else if (array_images.size() - 1 == j){
+                    j = 0;
+                    webView.loadUrl(array_images.get(j));
+                }
+                else {
+                    j += 1;
+                    webView.loadUrl(array_images.get(j));
+                    Log.v("Yes:", Integer.toString(j));
+                }
+            }
+        });
+
+        //If previous button is clicked, there are different scenarios that could happen:
+        //One: 0 images saved -> popup alert dialog again
+        //Two: j is at the first image of the list -> go to the last image of list
+        //Three: All other j index values -> load the (j-1)th image
+        previousbutton = (Button) rootView.findViewById(R.id.previous_image);
+        previousbutton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (array_images.size() == 0) {
+                    create_alert_search();
+                }
+                else if (j == 0){
+                    j = array_images.size()-1;
+                    webView.loadUrl(array_images.get(j));
+                }
+                else {
+                    j -= 1;
+                    webView.loadUrl(array_images.get(j));
+                    Log.v("Before:", Integer.toString(j));
+                }
+            }
+        });
+
+        //When the savebutton is clicked when there was no search, popup appears
+        //Else: get that particular image url and write it into the database using dbhelper
+        savebutton = (Button) rootView.findViewById(R.id.save_button);
+        savebutton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (array_images.size() == 0){
+                    create_alert_search();
+                }
+                else {
+                    dbhelper.writeData(array_images.get(j));
+                }
+            }
+        });
+
+        //When you want to switch layout back to view all images: Uses the main activity
+        //function "transitionToFragment", with the AllImagesFragment fragment
+        view_allbutton = (Button) rootView.findViewById(R.id.view_all_button);
+        view_allbutton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                AllImagesFragment fr = new AllImagesFragment();
+                ((MainActivity)getActivity()).transitionToFragment(fr);
+            }
+        });
+
+        return rootView;
+    }
```

Comment:
This method needs to be a lot shorter. you can make all of these onClicks more organized and independent of the method by using butterknife.